includes/content-distribution/class-distributor-migrator.php

```diff
@@ -0,0 +1,234 @@
+<?php
+/**
+ * Newspack Network Content Distribution Distributor Migrator.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Network\Content_Distribution;
+
+use Newspack_Network\Content_Distribution;
+use Newspack_Network\Utils\Network;
+use WP_Error;
+use WP_Query;
+use WP_Post;
+use InvalidArgumentException;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * Distributor Migrator Class.
+ */
+class Distributor_Migrator {
+	/**
+	 * Initialize hooks.
+	 */
+	public static function init() {
+		add_action( 'rest_api_init', [ __CLASS__, 'register_rest_routes' ] );
+	}
+
+	/**
+	 * Register REST routes.
+	 */
+	public static function register_rest_routes() {
+		register_rest_route(
+			'newspack-network/v1',
+			'/content-distribution/distributor-migrator/link/(?P<post_id>\d+)',
+			[
+				'methods'             => 'POST',
+				'callback'            => [ __CLASS__, 'api_link_incoming_post' ],
+				'args'                => [
+					'subscription_signature' => [
+						'type'     => 'string',
+						'required' => true,
+					],
+					'network_post_id'        => [
+						'type'     => 'string',
+						'required' => true,
+					],
+				],
+				'permission_callback' => '__return_true', // TODO: Check network signature.
```

Comment:
To complete the migration, there has to be a request from the distribution origin to its destination in order to set the network post ID, otherwise the migration will result in a duplicated post.

To secure that request, do we already have a single way that supports node->hub, hub->node, and node->node requests? cc @leogermani